rust/libnewsboat/src/utils.rs

```diff
@@ -477,6 +479,19 @@ pub fn getcwd() -> Result<PathBuf, io::Error> {
     env::current_dir()
 }
 
+/// Recursively create directories if missing and set permissions accordingly.
+pub fn mkdir_parents(p: &str, mode: u32) -> isize {
+    let path = Path::new(p);
+    let result = fs::DirBuilder::new()
+        .mode(mode)
+        .recursive(true) // directories created with same security and permissions
+        .create(path);
+    match result {
+        Ok(()) => 0,
+        Err(_) => -1,
```

Comment:
In C and C++, -1 indicates that there are an error, but the error itself is communicated out of band via `errno` variable—which this function doesn't set. Can you please adjust [the only place that checks `errno`](https://github.com/newsboat/newsboat/blob/c01acf47142637b8959036585bc30eaf28c30052/src/pbcontroller.cpp#L103-L110) to not use the variable? Just fail if ret is -1. It's safe not to check that errno is EEXIST because your test establishes that re-creating directories with this `make_parents` returns Ok.

Reply:
dropped the condition and removed errno completely incl. the log statement.